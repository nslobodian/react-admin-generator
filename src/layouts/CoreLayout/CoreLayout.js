import React from 'react'
import PropTypes from 'prop-types'
import LoadingBar from 'react-redux-loading-bar'
import Header from 'containers/HeaderContainer'
import 'styles/core.scss'

export const CoreLayout = ({ children }) => (
  <div>
    <Header />
    <LoadingBar className='loadingBar' />
    <div className='col-xs-12'>
      {children}
    </div>
  </div>
)

CoreLayout.propTypes = {
  children: PropTypes.element.isRequired,
}

export default CoreLayout
