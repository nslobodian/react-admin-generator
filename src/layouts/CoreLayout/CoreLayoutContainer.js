import connect from 'react-redux/es/connect/connect'
import CoreLayout from './CoreLayout'
import { logout } from 'redux/Auth'

const mapActionCreators = { logout }

const mapStateToProps = ({
  Auth: {
    loggedIn,
    admin: {
      username,
    },
  },
}) => ({
  loggedIn,
  username,
})

export default connect(mapStateToProps, mapActionCreators)(CoreLayout)
