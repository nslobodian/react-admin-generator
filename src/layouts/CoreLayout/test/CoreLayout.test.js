import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import CoreLayout from '../CoreLayout'

describe('(Layout) CoreLayout', () => {
  it('Should render CoreLayout properly', () => {
    const props = {
      children: [],
    }
    const wrapper = shallow(
      <CoreLayout
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
