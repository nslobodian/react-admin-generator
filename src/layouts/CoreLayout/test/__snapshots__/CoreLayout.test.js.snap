// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Layout) CoreLayout Should render CoreLayout properly 1`] = `
<div>
  <Connect(_class2) />
  <div
    className="col-xs-12"
  />
</div>
`;
